src/tpt_reports/tpt_reports.py

```diff
@@ -45,8 +52,85 @@ def load_json_file(file_path):
         return None
 
 
-# Issue #4 - Add ReportLab code and library
-# TODO: Add in the ReportLab code, library, parsing logic and remove this comment.
+# Issue #21 - test parse_json()
+# TODO: Add unit tests for following logic and remove this comment.
+def parse_json(data):
+    """Parse JSON object for values to report."""
+    payloads_list = []
+    payloads_meta = {}
+    num_payloads = 0
+    host_blocked = 0
+    host_not_blocked = 0
+    border_blocked = 0
+    border_not_blocked = 0
+    try:
+        if data:
+            for payload in data["payloads"]:
+                num_payloads += 1
+                payload_data = {}
+                payload_data["Payload"] = payload["payload_description"]
+                # c2_protocol requires additional space to match input file format
+                payload_data["C2 Protocol"] = payload["c2_protocol "]
```

Comment:
Is that extra trailing space meant to be there or should it be like this?
```suggestion
                payload_data["C2 Protocol"] = payload["c2_protocol"]
```

Reply:
The JSON data source that provides input to this program actually has the field as "c2_protocol ". Since the program that generates this data source is out of our control, I chose to just keep the field with a trailing space in our own program.

Reply:
Doh, I didn't understand what that comment meant, but now it makes perfect sense.  Thanks for pointing it out!